### PR TITLE
Nitant-Enhance Find

### LINKED ISSUE #47: As a user, I can find contacts by name and/or module code
....so that I can save time and narrow down my search.

### PR DESCRIPTION
**Pull request by nitant-p**
Find: Support searching by name and/or module

Find command can only search by name.

By allowing module searching, users can save time and find contacts easier depending on their motive. Users can also chain module codes and names to be more specific in their search.

Let's,
* update the search algorithm to tokenize the argument to search for specific fields
* update the name predicate class to accommodate for more fields
* update the test methods to verify the correctness of the changes

### CODE REVIEW

**Review by normkoh — COMMENTED** [review]
LGTM!

**Review by Jonaspng — APPROVED** [review]
LGTM.

File: src/main/java/seedu/address/model/person/PersonMatchesPredicate.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    public PersonMatchesPredicate() {
+        namesList = new ArrayList<>();
+        moduleList = new ArrayList<>();
+        hasNamesList = false;
+        hasModuleList = false;
+    }
+
+    @Override
+    public boolean test(Person person) {
```
Comment (GarethOng) [inline src/main/java/seedu/address/model/person/PersonMatchesPredicate.java:33]: Possible to rename ```test``` to fit Java coding standards.

**Review by GarethOng — COMMENTED** [review]
LGTM!

File: src/main/java/seedu/address/model/person/PersonMatchesPredicate.java (hunk 1 above)
Comment (nitant-p) [reply src/main/java/seedu/address/model/person/PersonMatchesPredicate.java:33]: Thanks for the comment. The `PersonMatchesPredicate` class implements the `Predicate` interface so it needs to override the `test` method.

**Review by JJoeYi — COMMENTED** [review]
LGTM

**Pull request merged**